server/routes.js

```diff
@@ -0,0 +1,254 @@
+var _fs             = require("fs")
+let denodeify       = require("denodeify")
+let config          = require("../config")
+let users           = require("../users")
+
+const fs = {
+	readFile: denodeify(_fs.readFile)
+}
+
+module.exports = [
+	{
+		path:/^\/$/,
+		page: "page.hbs",
+		cache: true,
+		prerender: ["/"],
+		context: (_, db) =>
+			aggregatePosts(db, { $sort: { timestamp: -1 } }, { $limit: 5 })
+				.then(fillAuthorInfo(db))
+				.then((posts) => {
+					posts.forEach((post) => post.short = true);
+					return db.posts.count()
+						.then((count) => ({
+							title: "",
+							posts,
+							pagination: {
+								right: count > 5 ? "/archive/2" : undefined
+							}
+						}))
+				})
+				.then(fillDefaultSidebar(db))
+	},
+	{
+		path:/^\/archive\/([1-9][0-9]*)$/,
+		page: "page.hbs",
+		cache: true,
+		prerender: range(5).map((i) => `/archive/${i + 1}`),
+		context: ([_, pageNumber], db) => {
+			pageNumber = parseInt(pageNumber);
+			return aggregatePosts(db, { $sort: { timestamp: -1 } }, { $skip: (pageNumber - 1) * 5 }, { $limit: 5 })
+				.then(fillAuthorInfo(db))
+				.then((posts) => {
+					posts.forEach((post) => post.short = true);
+					return db.posts.count()
+						.then((count) => ({
+							title: "",
+							posts,
+							pagination: {
+								left: pageNumber > 1 ? `/archive/${pageNumber - 1}` : undefined,
+								right: count > pageNumber * 5 ? `/archive/${pageNumber + 1}` : undefined
+							}
+						}))
+				})
+				.then(fillDefaultSidebar(db))
+		}
+	},
+	{
+		path:/^\/editor(\/([0-9a-zA-Z_-]{7,14}))?$/,
+		page: "editor.hbs",
+		cache: false,
+		context: ([_, __, postId], db) =>
+			db.posts.findOne({ guid: postId })
+				.then((post) => {
+					if (post !== undefined) {
+						return {
+							title: `Editing "${post.title}"`,
+							post,
+							admin: true
+						}
+					} else {
+						return {
+							title: `Editing New Post`,
+							admin: true
+						}
+					}
+				})
+	},
+	{
+		path:/^\/tag\/([a-z0-9\-]{1,16})$/,
+		page: "page.hbs",
+		cache: true,
+		prerender: (db) => {
+			return db.posts.distinct("tags", {})
+				.then((tags) => tags.map((tag) => `/tag/${tag}`))
+		},
+		context: ([_, tag], db) =>
+			aggregatePosts(db, { $match: { tags: tag } }, { $sort: { timestamp: -1 } })
+				.then(fillAuthorInfo(db))
+				.then((posts) => {
+					posts.forEach((post) => post.short = true);
+					return {
+						title: `Tagged "${tag}"`,
+						posts: posts
+					};
+				})
+				.then(fillDefaultSidebar(db))
+	},
+	{
+		path:/^\/author\/(matt|zach|ben|danial|david|ellis|kyle|thomas)$/, // sigh...
```

Comment:
yeah, with the new format it doesn't make sense. That was originally structured like that as a convenience, but with so much being done by the renderer now we should probably change it